lib/sqlalchemy/orm/mapper.py

```diff
@@ -1262,7 +1263,7 @@ def _configure_class_instrumentation(self):
                 method = method._sa_original_init
                 if isinstance(method, types.MethodType):
                     method = method.__func__
-            if isinstance(method, types.FunctionType):
+            if isinstance(method, types.FunctionType) or inspect._signature_is_functionlike(method):
```

Comment:
`_signature_is_functionlike` is a private function. Is there a way of using a non-private method?

Reply:
agree this cannot be accepted using private python API

Reply:
Another solution to this problem is to copy called function to one of the modules in the SQLAlchemy

Reply:
sure, but if there is an alternative that does not need it it's preferable

In my option the issue is in part on the cython side

Reply:
Could we assume that any `callable` is a valid validator? If yes, then the or statement could just contain `or callable(method)`.

Reply:
I guess we can just us duck typing here:
```
if isinstance(method, types.MethodType): -> if hasattr(method, '__func__'):
if isinstance(method, types.FunctionType): -> if callable(method):
```